stdlib/public/Reflection/TypeLowering.cpp

```diff
@@ -1034,7 +1034,11 @@ class ExistentialTypeInfoBuilder {
           ++WitnessTableCount;
 
           if (auto *Superclass = TC.getBuilder().lookupSuperclass(P)) {
-            auto *SuperclassTI = TC.getTypeInfo(Superclass);
+            // ObjC class info should be available in the metadata, so it's safe
+            // to instantiate a dummy reader here. This helps preserving the
+            // layering.
+            remote::MemoryReader dummyReader;
+            auto *SuperclassTI = TC.getTypeInfo(Superclass, dummyReader);
```

Comment:
Could the reader parameter to getTypeInfo() be a pointer instead, with handing of the null case?

Reply:
Maybe llvm:Optional to make it clear the reader is... well... optional?

Reply:
I believe llvm::Optional also has the benefit of being able to keep `queryDataLayout`, `getSymbolAddress` and `readString` deleted in the base class.